### ISSUE TITLE
crash on killEntitiesOnEnd

### ISSUE THREAD

**Issue by arketec** [arketec/Custom-Machinery#5]
per discord:
https://discord.com/channels/842550366763614258/1259637197381046312/1282231694979502134

**Comment by arketec** [comment]
from Discord
```
dozingpip
 — 
09/07/2024 11:50 PM
o/ I got this crash from running:

ServerEvents.recipes(event => {
    event.recipes.custommachinery.custom_machine("playingwithfire:enderman_spawner", 200)
    .requireBlock("minecraft:soul_sand", true, -1, -1, -1, 1, 1, 1, 1)
    .runCommandOnEnd("/summon minecraft:enderman ~ ~1 ~")
    .killEntitiesOnEnd(1, 10, ["minecraft:zombified_piglin", "minecraft:piglin"], true)
})



it works fine and does not crash if I use requireEntities instead of killEntitiesOnEnd:

ServerEvents.recipes(event => {
    event.recipes.custommachinery.custom_machine("playingwithfire:enderman_spawner", 200)
    .requireBlock("minecraft:soul_sand", true, -1, -1, -1, 1, 1, 1, 1)
    .runCommandOnEnd("/summon minecraft:enderman ~ ~1 ~")
    .requireEntities(1, 10, ["minecraft:zombified_piglin", "minecraft:piglin"], true)
})


I get a similar crash with consumeEntityHealthOnEnd
I tried in 1.21 with just cm and kubejs, same crash (1.20 is the main version I'm working with, but I was curious)
```

**Comment by arketec** [comment]
```
---- Minecraft Crash Report ----
// Quite honestly, I wouldn't worry myself about that.

Time: 2024-09-08 02:17:38
Description: Ticking block entity

java.lang.IllegalStateException: Missing key in ResourceKey[minecraft:root / minecraft:damage_type]: ResourceKey[minecraft:damage_type / custommachinery:machine_damage]
at net.minecraft.core.Registry.m_257106_(Registry.java:151) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:classloading,re:mixin}
at java.util.Optional.orElseThrow(Unknown Source) ~[?:?] {re:mixin}
at net.minecraft.core.Registry.m_246971_(Registry.java:151) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:classloading,re:mixin}
at fr.frinn.custommachinery.common.util.CustomMachineDamageSource.<init>(CustomMachineDamageSource.java:17) ~[CustomMachinery-forge-1.20.1-0.10.5.jar%23203!/:?] {re:classloading}
at fr.frinn.custommachinery.common.component.EntityMachineComponent.lambda$new$0(EntityMachineComponent.java:26) ~[CustomMachinery-forge-1.20.1-0.10.5.jar%23203!/:?] {re:classloading}
at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:183) ~[guava-31.1-jre.jar%23118!/:?] {}
at fr.frinn.custommachinery.common.component.EntityMachineComponent.lambda$killEntities$6(EntityMachineComponent.java:72) ~[CustomMachinery-forge-1.20.1-0.10.5.jar%23203!/:?] {re:classloading}
at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source) ~[?:?] {}
at java.util.stream.SliceOps$1$1.accept(Unknown Source) ~[?:?] {}
at java.util.ArrayList$ArrayListSpliterator.tryAdvance(Unknown Source) ~[?:?] {}
at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source) ~[?:?] {}
at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source) ~[?:?] {}
at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?] {}
at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?] {}
at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source) ~[?:?] {}
at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source) ~[?:?] {}
at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?] {}
at java.util.stream.ReferencePipeline.forEach(Unknown Source) ~[?:?] {}
at fr.frinn.custommachinery.common.component.EntityMachineComponent.killEntities(EntityMachineComponent.java:72) ~[CustomMachinery-forge-1.20.1-0.10.5.jar%23203!/:?] {re:classloading}
at fr.frinn.custommachinery.common.requirement.EntityRequirement.processEnd(EntityRequirement.java:130) ~[CustomMachinery-forge-1.20.1-0.10.5.jar%23203!/:?] {re:classloading}
at fr.frinn.custommachinery.common.requirement.EntityRequirement.processEnd(EntityRequirement.java:28) ~[CustomMachinery-forge-1.20.1-0.10.5.jar%23203!/:?] {re:classloading}
at fr.frinn.custommachinery.common.crafting.machine.MachineProcessor.endProcess(MachineProcessor.java:204) ~[CustomMachinery-forge-1.20.1-0.10.5.jar%23203!/:?] {re:classloading}
at fr.frinn.custommachinery.common.crafting.machine.MachineProcessor.tick(MachineProcessor.java:89) ~[CustomMachinery-forge-1.20.1-0.10.5.jar%23203!/:?] {re:classloading}
at fr.frinn.custommachinery.common.init.CustomMachineTile.serverTick(CustomMachineTile.java:267) ~[CustomMachinery-forge-1.20.1-0.10.5.jar%23203!/:?] {re:classloading}
at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.m_142224_(LevelChunk.java:689) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:classloading}
at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.m_142224_(LevelChunk.java:782) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:classloading}
at net.minecraft.world.level.Level.m_46463_(Level.java:468) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:LevelMixin,pl:mixin:APP:botania_xplat.mixins.json:LevelAccessor,pl:mixin:A}
at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:351) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:libx:level_load,re:classloading,pl:accesstransformer:B,xf:fml:libx:level_load,pl:mixin:APP:kubejs-common.mixins.json:ServerLevelMixin,pl:mixin:APP:botania_xplat.mixins.json:ServerLevelMixin,pl:mixin:APP:create.mixins.json:accessor.ServerLevelAccessor,pl:mixin:A}
at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:893) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:814) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
at net.minecraft.client.server.IntegratedServer.m_5705_(IntegratedServer.java:89) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:classloading,pl:runtimedistcleaner:A}
at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:661) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:251) ~[client-1.20.1-20230612.114412-srg.jar%23230!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
at java.lang.Thread.run(Unknown Source) ~[?:?] {}

```

**Comment by arketec** [comment]
fixed f7263e06041750b203290a1ea319bc90e72082fc
